File: src/main/java/org/radarcns/config/MonitorConfig.java
package org.radarcns.config;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * POJO representing a monitor configuration
 */
public class MonitorConfig {
    @JsonProperty("email_address")
    private String emailAddress;

    @JsonProperty("email_host")
    private String emailHost;

    @JsonProperty("email_port")
    private int emailPort;

    @JsonProperty("email_user")
    private String emailUser;

    private List<String> topics;

    public String getEmailAddress() {
        return emailAddress;
    }

    public void setEmailAddress(String emailAddress) {
        this.emailAddress = emailAddress;
    }

    public List<String> getTopics() {
        return topics;
    }

    public void setTopics(List<String> topics) {
        this.topics = topics;
    }

    public String getEmailHost() {
        return emailHost;
    }

    public void setEmailHost(String emailHost) {
        this.emailHost = emailHost;
    }

    public int getEmailPort() {
        return emailPort;
    }

    public void setEmailPort(int emailPort) {
        this.emailPort = emailPort;
    }

    public String getEmailUser() {
        return emailUser;
    }

    public void setEmailUser(String emailUser) {
        this.emailUser = emailUser;
    }
}

File: src/main/kotlin/org/radarbase/kotlin/config/MonitorConfig.kt
package org.radarbase.kotlin.config

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * POJO representing a monitor configuration
 */
class MonitorConfig {
    @JsonProperty("notify")
    var notifyConfig: List<NotifyConfig> = emptyList()

    @JsonProperty("email_host")
    var emailHost: String = ""

    @JsonProperty("email_port")
    var emailPort: Int = 0

    @JsonProperty("email_user")
    var emailUser: String = ""

    @JsonProperty("log_interval")
    var logInterval: Int = 1000

    var topics: List<String>? = null

    @JsonProperty("message")
    var message: String? = null
}